### PR TITLE
vendor: bump pebble to 3b86a4009c3b

### PR DESCRIPTION
**Pull request by petermattis**
* sstable: don't skip over errored blocks during iteration
* db: avoid go1.14 unsafe pointer conversion violations
* db: refactor compaction startLevel, outputLevel
* db: allow multiple levels in CompactionInfo.Input

Release note (bug fix): Fix a bug where Pebble iteration would skip over 
blocks which failed checksum verification rather than propagating the error
to the caller.

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -93,14 +93,14 @@ Run `go get -u <dependency>`. To get a specific version, run `go get -u <depende
 When updating a dependency, you should run `go mod tidy` after `go get` to ensure the old entries
 are removed from go.sum.
 
-You must then run `make -k vendor_rebuild` to ensure the modules are installed. These changes must
```
Comment (otan) [inline build/README.md:96]: you still need the `-k` if your vendor gets into a weird state and execgen won't run.

File: go.sum [hunk 2]
```diff
@@ -147,6 +147,8 @@ github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f h1:o/kfcElHqOi
 github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f/go.mod h1:i/u985jwjWRlyHXQbwatDASoW0RMlZ/3i9yJHE2xLkI=
 github.com/cockroachdb/pebble v0.0.0-20200624214307-560ed0b8a18c h1:MQVYGjMEPj6TRGTi05ct75tDhbBKKFoJ0yKvvEZzIZw=
 github.com/cockroachdb/pebble v0.0.0-20200624214307-560ed0b8a18c/go.mod h1:crLnbSFbwAcQNs9FPfI1avHb5BqVgqZcr4r+IzpJ5FM=
```
Comment (otan) [inline go.sum:149]: missing a `go mod tidy`

**Review by otan — APPROVED** [review]

File: go.sum (hunk 2 above)
Comment (otan) [reply go.sum:149]: (instruction for that included if you rebase i think!)

File: build/README.md (hunk 1 above)
Comment (petermattis) [reply build/README.md:96]: Really? `make vendor_rebuild` shouldn't be building `execgen`.

File: go.sum (hunk 2 above)
Comment (petermattis) [reply go.sum]: Done.

**Comment by itsbilal** [comment]
Random question: why is it that every bump of pebble also pulls in a new version of `x/sys`?

File: build/README.md (hunk 1 above)
Comment (petermattis) [reply build/README.md:96]: I just tested this:

```
~ rm bin/execgen*
~ make vendor_rebuild
...
~ ls bin/execgen*
ls: bin/execgen*: No such file or directory
```

Comment (otan) [reply build/README.md:96]: weird, it does that on my system.

```
make vendor_rebuild
Running make with -j8
GOPATH set to /Users/otan/go
go install -v execgen
bin/prereqs ./pkg/sql/colexec/execgen/cmd/execgen > bin/execgen.d.tmp
```

Comment (petermattis) [reply build/README.md:96]: Are you running on top of https://github.com/cockroachdb/cockroach/pull/50636?

**Comment by petermattis** [comment]
> Random question: why is it that every bump of pebble also pulls in a new version of x/sys?

Because that repo changes super frequently. I agree this is sort of irritating. Perhaps we should pin that dep.

**Comment by otan** [comment]
> Because that repo changes super frequently. I agree this is sort of irritating. Perhaps we should pin that dep.

i think it is pinned, just `go get` decides it likes to upgrade it every time it's run :\

File: build/README.md (hunk 1 above)
Comment (otan) [reply build/README.md:96]: ah, nope!

**Comment by petermattis** [comment]
TFTR!

bors r+

**Pull request merged**